subprojects/core/src/main/java/org/gradle/groovy/scripts/internal/RegistryAwareClassLoaderHierarchyHasher.java

```diff
@@ -32,14 +33,11 @@ public RegistryAwareClassLoaderHierarchyHasher(ClassLoaderRegistry registry, Has
     private static Map<ClassLoader, String> collectKnownClassLoaders(ClassLoaderRegistry registry) {
         Map<ClassLoader, String> knownClassLoaders = Maps.newHashMap();
 
-        String javaVmVersion = String.format("%s|%s", System.getProperty("java.vm.name"), System.getProperty("java.vm.vendor"));
-        ClassLoader systemClassLoader = ClassLoader.getSystemClassLoader();
-        if (systemClassLoader != null) {
-            addClassLoader(knownClassLoaders, systemClassLoader, "system-app" + javaVmVersion);
-            addClassLoader(knownClassLoaders, systemClassLoader.getParent(), "system-ext" + javaVmVersion);
-        }
-
         String gradleVersion = GradleVersion.current().getVersion();
+
+        // Some implementations may use null to represent the bootstrap class loader and in such cases
```

Comment:
Is this _some_ or _all_ implementations? I think it's the latter..?

Reply:
The documentation says

> Returns the parent class loader for delegation. Some implementations may use null to represent the bootstrap class loader. This method will return null in such implementations if this class loader's parent is the bootstrap class loader.